File: scripts/spline/modular/postprocess.jl
using DrWatson
@quickactivate :ens

using RData
using DataFrames
using DataFramesMeta
using Spikes
using Arrow

include(srcdir("spline", "model_summaries.jl"))

# inpath = "/home/ginko/ens/data/analyses/spline/batch-test/best-neigh/post-proc/fit.RData"

inpath = ARGS[1]
outpath = ARGS[1]


data = RData.load(inpath)["predictions"]

result = DataFrame(index1=Float64[],
				   index2=Float64[], 
				   group=String[],
				   reference=String[],
				   landmark=String[],
				   variable=String[],
				   x=Vector{Float64}[],
				   mean=Vector{Float64}[],
				   sd=Vector{Float64}[])

for row in data
	variables  = filter(x -> (x != "index1" && 
							  x != "index2" &&
							  x != "group" &&
							  x != "reference" &&
							  x != "landmark"), 
						row.index2name)

	index1 = row["index1"]
	index2 = ismissing(row["index2"]) ? NaN : row["index2"]
	group = row["group"]
	reference = row["reference"]
	landmark = row["landmark"]

	for v in variables
		push!(result, (index1, index2, group, reference, landmark, v, row[v]["x"], row[v]["mean"], row[v]["sd"]))
	end
end

Arrow.write(outpath, result)

File: scripts/spline/modular/postprocess.R
library(STAR)

args <- commandArgs(trailingOnly = TRUE)
inpath = paste(args[1], "out", "data", sep="/")
outpath = args[2]

read_rdata <- function(file) {
	r_data <- load(file)
	x <- get(r_data)
	return(x)
}

predict.one.var <- function(result, var) {
	pred = quickPredict(result$gss, var)

	x = pred$xx
	mean = pred$est.mean
	sd = pred$est.sd

	if (is.function(result$inv.rnfun[[var]])) {
		x = result$inv.rnfun[[var]](x)
	}

	return(list(x=x, mean=mean, sd=sd))
}

predict.one.result <- function(file) {
	result = read_rdata(file)
	time = "time"
	if (result$reference == "multi") {
		time = "timetoevt"
	}

	vars = c(time, "r.timeSinceLastSpike")
	if (result$group != "all") {
		vars <-c(vars, "r.nearest")
	}


	res = list(index1=result$index1, index2=result$index2, reference=result$reference, group=result$group, landmark=result$landmark)

	for (var in vars) {
		res[[var]] = predict.one.var(result, var)
		}
	return(res)
	}

message("\nLoading files from: ", inpath)
infiles <- list.files(path=inpath, pattern=".R", full.names=T, all.files=T)


predictions = lapply(infiles, predict.one.result)

save(predictions, file=outpath)
